Modules/IO/Meta/test/itkLargeMetaImageWriteReadTest.cxx

```diff
@@ -73,17 +71,13 @@ ActualTest(std::string filename, typename TImageType::SizeType size)
 
     std::cout << "Initializing pixel values " << std::endl;
 
-    IteratorType itr(image, region);
-    itr.GoToBegin();
-
-    pixelValue = itk::NumericTraits<PixelType>::ZeroValue();
-
+    PixelType pixelValue = itk::NumericTraits<PixelType>::ZeroValue();
     chronometer.Start("Initializing");
-    while (!itr.IsAtEnd())
+    for (IteratorType itr(image, region); !itr.IsAtEnd(); ++itr)
     {
       itr.Set(pixelValue);
-      ++pixelValue;
-      ++itr;
+      // NOTE: pixelValue for unsigned short will wrap around with large images.
+      pixelValue = (pixelValue != itk::NumericTraits<PixelType>::max()) ? ++pixelValue : 0;
```

Comment:
It is likely that the GCC warning `operation on ... may be undefined [-Wsequence-point]` will not appear if the expression is

```cpp
pixelValue = (pixelValue != itk::NumericTraits<PixelType>::max()) ? pixelValue + 1 : 0;
```

or

```cpp
if (pixelValue != itk::NumericTraits<PixelType>::max())
{
  ++pixelValue;
}
else
{
  pixelValue = 0;
}
```

This warning happens with _-Wall_ argument, for some reason _GCC_ don't like expressions like:


```cpp
int main(int, char**)
{
	int x = 1;
	x = ++x;
	return 0;
}
```
```
r@deb2:~$ g++ -Wall test1.cpp 
test1.cpp: In function ‘int main(int, char**)’:
test1.cpp:4:11: warning: operation on ‘x’ may be undefined [-Wsequence-point]
    4 |         x = ++x;
      |         ~~^~~~~
```